addons/account/views/account_move_views.xml

```diff
@@ -946,9 +947,9 @@
                             <div class="text-warning" invisible="not show_name_warning">The current highest number is <field class="oe_inline" name="highest_name"/>. You might want to put a higher number here.</div>
 
                             <h1>
-                                <field name="name" invisible="name == '/' and not posted_before and not quick_edit_mode" readonly="state != 'draft'" placeholder="Draft"/>
+                                <field name="name" invisible="not ((state != 'draft') or name_placeholder_form or (name and (posted_before or quick_edit_mode)))" readonly="state != 'draft'" options="{'placeholder_field': 'name_placeholder_form'}"/>
```

Comment:
This looks complicated.
Since `placeholder_name` should only be computed when a placeholder is needed, I think this should be enough.

```suggestion
                                <field name="name" invisible="not (name or placeholder_name or quick_edit_mode)" readonly="state != 'draft'" options="{'placeholder_field': 'name_placeholder_form'}"/>
```